python-lib/text_extraction/__init__.py

```diff
@@ -57,57 +58,57 @@ def extract_text_content(file_bytes, extension, with_pandoc):
 
 
 def extract_text_chunks(filename, file_bytes, extension, with_pandoc, metadata_as_plain_text, use_pdf_bookmarks):
-    if extension == "pdf":
-        pdf_pages = pdfium.PdfDocument(file_bytes)
-        outline = list(pdf_pages.get_toc())
-        if len(outline) == 0 or not use_pdf_bookmarks:
-            # only extract page numbers when no outline is found
-            return [
-                {
-                    'file': filename,
-                    'text': page.get_textpage().get_text_range(),
-                    'chunk_id': page_id + 1,
-                    'metadata': "Page {}".format(page_id + 1) if metadata_as_plain_text else {"page": page_id + 1},
-                    'error_message': ""
-                }
-                for page_id, page in enumerate(pdf_pages)
-            ]
-        else:
-            return _extract_pdf_chunks(filename, pdf_pages, outline, metadata_as_plain_text)
-    elif extension == "doc":
+    if extension == "doc":
         raise ValueError("'doc' files are not supported, try to convert them to docx.")
-    elif extension == "md":
-        return _extract_markdown_chunks(file_bytes.decode(), filename, metadata_as_plain_text)
-    else:
-        try:
-            if not with_pandoc:
-                raise ValueError("pandoc is required to extract chunks from files (except for PDFs and markdown).")
 
+    try:
+        if extension == "pdf":
+            pdf_pages = pdfium.PdfDocument(file_bytes)
+            outline = list(pdf_pages.get_toc())
+            if len(outline) == 0 or not use_pdf_bookmarks:
+                # only extract page numbers when no outline is found
+                return [
+                    {
+                        'file': filename,
+                        'text': page.get_textpage().get_text_range(),
+                        'chunk_id': page_id + 1,
+                        'metadata': "Page {}".format(page_id + 1) if metadata_as_plain_text else {"page": page_id + 1},
+                        'error_message': ""
+                    }
+                    for page_id, page in enumerate(pdf_pages)
+                ]
+            else:
+                return _extract_pdf_chunks(filename, pdf_pages, outline, metadata_as_plain_text)
+        elif extension == "md":
+            if not with_pandoc:
+                raise ValueError("pandoc is required to extract chunks from markdown files.")
```

Comment:
should we make it work for "md" file without pandoc ? Using the `pypandoc.convert_text` method depending on an args of the `_extract_markdown_chunks` method ?

Reply:
https://github.com/dataiku/dss-plugin-tesseract-ocr/pull/76/commits/50c1285f57609a404f941270c11deacfd1835cd8